packages/components/src/components/modal/Modal.tsx

```diff
@@ -111,7 +115,10 @@ const Modal: React.FC<ModalProps> = ({
           document.addEventListener(EventListener.keydown, onKeyDown);
 
           // Cleanup event listener
-          return () => document.removeEventListener(EventListener.keydown, onKeyDown);
+          return () => {
+            document.removeEventListener(EventListener.keydown, onKeyDown);
+            previousFocusedElement?.focus();
```

Comment:
could have weird behaviours if focusTrapEnabled is switched from true to false (while show stays true) though I am not sure it's a realistic scenario